cylc/uiserver/schema.py

```diff
@@ -248,14 +254,205 @@ class Arguments:
     result = GenericScalar()
 
 
+async def get_jobs(root, info, **args):
+    if args['live']:
+        return await get_nodes_all(root, info, **args)
+
+    _, field_ids = process_resolver_info(root, info, args)
+
+    if hasattr(args, 'id'):
+        args['ids'] = [args.get('id')]
+    if field_ids:
+        if isinstance(field_ids, str):
+            field_ids = [field_ids]
+        elif isinstance(field_ids, dict):
+            field_ids = list(field_ids)
+        args['ids'] = field_ids
+    elif field_ids == []:
+        return []
+
+    for arg in ('ids', 'exids'):
+        # live objects can be represented by a universal ID
+        args[arg] = [Tokens(n_id, relative=True) for n_id in args[arg]]
+    args['workflows'] = [
+        Tokens(w_id) for w_id in args['workflows']]
+    args['exworkflows'] = [
+        Tokens(w_id) for w_id in args['exworkflows']]
+
+    return await list_jobs(args)
+
+
+async def list_jobs(args):
+    if not args['workflows']:
+        raise Exception('At least one workflow must be provided.')
+    from cylc.flow.workflow_db_mgr import WorkflowDatabaseManager
+    from pathlib import Path
+    from cylc.flow.workflow_files import get_workflow_srv_dir
+
+    jobs = []
+    for workflow in args['workflows']:
+        db_file = Path(get_workflow_srv_dir(workflow['workflow']), 'db')
+        wdbm = WorkflowDatabaseManager(db_file.parent)
+        with wdbm.get_pri_dao() as pri_dao:
+            # TODO: support all arguments including states
+            for row in pri_dao.connect().execute('''
```

Comment:
It might be worth splitting the code from this line down into another function e.g:

```python
    ...
    with wdbm.get_pri_dao() as pri_dao:
        return query_stats(pri_dao)

def query_stats(pri_dao):
    ...
```

So that the query / response can be tested without having to go through all the Cylc/GraphQL machinery something like this:

```python
# cylc/uiserver/tests/test_stats.py or whatnot

def _get_db(values):
    # untested
    db_file = tmp_path / 'db'
    conn = sqlite3.connect(db_file)
    conn.execute('CREATE TABLE <table> ...')
    conn.executemany('INSERT INTO <table> VALUES (?,?,?...)', values) 
    conn.execute()
    return conn

def test_query_stats():
    conn = _get_db([
        'task1', cycle1', '01', ...,
        'task1', cycle2', '01', ...,
    ])
    stats = query_stats(conn)
    assert stats[0]['max_queue_time'] = ...
```